[WIP]Add skip to content- keyboard accessibility

Aditya-Kolla opened the pull request:
<!-- If this is related to a Bugzilla bug, please begin your title with [Bug XXXXX] and update the following link. -->
<!-- If there is no bug, consider making one and if you really don't want one, remove the link. -->

<!-- Include any other context you wish here. -->
This is a WIP PR.
Temporary position and look.
Created a new component for SkipNavigation.
fixes #314

Aditya-Kolla commented:
@helfi92 Can you help me understand the styling a bit. 
I have figured it out on plain css but not able to translate it to our project.

helfi92 commented on ui/src/components/SkipNavigation/index.jsx:
```diff
@@ -0,0 +1,15 @@
+    const { href, text } = this.props;
+
+    return (
+      <Link to={href}>
```
I think you can use a Button from `src/components/Button` instead and have an `onClick` that jumps to the first focusable element inside `<main>`.

Aditya-Kolla replied:
Alright. Will do.

Aditya-Kolla replied:
I have done these changes.

Aditya-Kolla commented:
@helfi92 I was thinking of pushing it out of the screen space but the clipping method which you showed seems like a better idea. I'll implement that :)

helfi92 commented:
> I was thinking of pushing it out of the screen space but the clipping method which you showed seems like a better idea. I'll implement that :)

The clipping idea is not mine. I have never used it. We can try using it if it makes sense :)

Aditya-Kolla commented:
> The clipping idea is not mine. I have never used it. We can try using it if it makes sense :)

Ah, fair enough.

Aditya-Kolla commented:
@helfi92 I've changed the Link to Button and added css to make it invisible by default and appear when it gets focus. 
I realised that to use clip I need to define a canvas so implemented my initial approach.
Yet to add button onClick which I'll be doing as soon as I wake up tomorrow morning :sleeping:

helfi92 commented:
Awesome, good night :)

Aditya-Kolla commented:
Hey @helfi92, I started work on the onClick and was wondering how could I get the first focus-sable element in the main tag? 
Will I need to use refs?

helfi92 commented:
> I started work on the onClick and was wondering how could I get the first focus-sable element in the main tag

I think it should be fine to simply jump to the `<main>` element. You can use the DOM API to have access. Maybe something like `document.querySelector('main')`.

Aditya-Kolla commented:
Hello, sorry for not being able to work on this lately. 
In this commit, I changed the skip button color to red so it brings a little contrast.
And the focus changes to main tag on button click.
Please review.

djmitche commented:
Hassan will be back next week and can look at this.

helfi92 commented:
@Aditya-Kolla How is it going? Let me know if you had any questions for me :)

Aditya-Kolla commented:
> @Aditya-Kolla How is it going? Let me know if you had any questions for me :)

Hello, sorry I got a bit busy. I'll resume work on this :D

helfi92 commented on ui/src/components/SkipNavigation/index.jsx:
```diff
@@ -0,0 +1,57 @@
+@withStyles({
+  skipButton: {
+    position: 'absolute',
+    left: '-999px',
```
`left: -999,`

Aditya-Kolla replied:
Done. Mind if I ask what the difference between the two is? Thank you :D

helfi92 replied:
There is no difference except fewer characters and consistency througout the codebase. :)

helfi92 commented:
@Aditya-Kolla I accidentally pushed a commit. Do you mind creating a new PR with your changes, please. Sorry for that.
